From 0f956c3bd8bb9f70ed4b8db67a5a22b7020a5f82 Mon Sep 17 00:00:00 2001
From: Mohamed Heikal <mheikal@chromium.org>
Date: Tue, 4 Apr 2023 17:04:04 +0000
Subject: [PATCH] Shorted printed cmd when long commands fail

Some commands (eg: javac) are very long and when they fail most of the
screen/log is filled with the actual cmd vs the error message output.
Shorten printed CMDs to 200 chars unless an environment variable is set.

Bug: None
Change-Id: I92809a7e4a2b1204931a74fd6239803feddd430e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4395905
Reviewed-by: Andrew Grieve <agrieve@chromium.org>
Commit-Queue: Mohamed Heikal <mheikal@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1126065}
Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Backport [114.0.5696.0 4c6fc1984970af4b2b1765014c9ddcd957ad7dda +import removal from 116.0.5809.0 4ba408349953db7b56a81ed246ef0ff03c5512a6]

 src/build/android/gyp/util/build_utils.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/build/android/gyp/util/build_utils.py b/src/build/android/gyp/util/build_utils.py
index af19f14fb6..7c778ecb7f 100644
--- a/src/build/android/gyp/util/build_utils.py
+++ b/src/build/android/gyp/util/build_utils.py
@@ -12,7 +12,6 @@ import fnmatch
 import json
 import logging
 import os
-import pipes
 import re
 import shlex
 import shutil
@@ -196,9 +195,14 @@ class CalledProcessError(Exception):
 
   def __str__(self):
     # A user should be able to simply copy and paste the command that failed
-    # into their shell.
+    # into their shell (unless it is more than 200 chars).
+    # User can set PRINT_FULL_COMMAND=1 to always print the full command.
+    print_full = os.environ.get('PRINT_FULL_COMMAND', '0') != '0'
+    full_cmd = shlex.join(self.args)
+    short_cmd = textwrap.shorten(full_cmd, width=200)
+    printed_cmd = full_cmd if print_full else short_cmd
     copyable_command = '( cd {}; {} )'.format(os.path.abspath(self.cwd),
-        ' '.join(map(pipes.quote, self.args)))
+                                              printed_cmd)
     return 'Command failed: {}\n{}'.format(copyable_command, self.output)
 
 
